### PR TITLE
Version bump zlib/1.2.12 in doxygen packages

### PR DESCRIPTION
**Pull request by vdsbenoit**
Specify library name and version:  **doxygen/***

Version bump zlib dependency to version 1.2.12

Since to zlib update to 1.2.12, using the doxygen package throws a diamond conflict with boost. 

image

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by madebr — APPROVED** [review]

**Comment by vdsbenoit** [comment]
@madebr, do you know if anyone from the Conan Center team could merge this PR? As explained in [this comment](https://github.com/conan-io/conan/issues/7016#issuecomment-1104126382), the new version of zlib has broken our pipelines. And I believe it might have broken some more too.

**Comment by madebr** [comment]
This pr is not acceptable in its current state because you, or the author of the patch, has not requested access in #4 yet.
Weird, because github recognized you as a contributor.
Have you left a message at #4 requesting access?
Is this commit created by a mail address connected to your github account?

**Comment by vdsbenoit** [comment]
> Weird, because github recognized you as a contributor to this repo.

Maybe because I have already contributed to issues & PR 🤷‍♂️

> Have you left a message at https://github.com/conan-io/conan-center-index/issues/4 requesting access?
No, I did not requested access on purpose because I did not intended on becoming a maintainer of the conan index. 

I just created this PR instead of an issue to speed up the process of fixing the broken state of Conan Center. 

> Is this commit created by a mail address connected to your github account?

I think so because I made it through the web GUI.

**Comment by madebr** [comment]
> > Weird, because github recognized you as a contributor to this repo.
> 
> Maybe because I have already contributed to issues & PR man_shrugging
> 
> > Have you left a message at #4 requesting access?
> 
> No, I did not requested access on purpose because I did not intended on becoming a maintainer of the conan index.

#4 is there to block bots and abuse, not a sign-up for maintainers.
The bot will only build for people who have "access".
So without #4, this pr will never get a green light.

I pinkie promise I will not come after you for breaking recipes in this repo :stuck_out_tongue_winking_eye:  :rofl:

**Comment by uilianries** [comment]
@vdsbenoit we described it here: https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md#request-access 

Indeed we had few spam PRs in the past, so the bot saved us. Also, the issue #4 is intended to be a welcome thread, we had the idea about people introducing themself as first contributor (similar to web forums).

**Comment by vdsbenoit** [comment]
All right, I've just added a post on #4. 

This PR will likely be blocked by https://github.com/conan-io/conan-center-index/pull/10470 though...

**Comment by madebr** [comment]
You probably saw a message along the following line:
```
ERROR: Conflict in depc/0.1:
    'depc/0.1' requires 'depa/0.1' while 'depb/0.1' requires 'depa/0.2'.
    To fix this conflict you need to override the package 'depa' in your root package.
```

You can fix the conflict by doing as the message says: explicitly add the problem dependency to your requirements.
In this case, you'ld need to add `zlib/1.2.12` to the requirements list of your particular recipe.
You will probably also need to add `--build missing` when building the conan packages.

**Comment by madebr** [comment]
You can auto-update the locks by adding `--lockfile conan.lock --lockfile-out conan.lock` to `conan create`.
Using these options, I think you can simulate the behavior of `npm ci`.
I guess for conan 1.x this is what you're going to have to live with.

I don't have much experience with lock files, so I can't really help you further.

**Comment by vdsbenoit** [comment]
Unfortunately, it is not that easy. I think we need to define base lock files for every profile first

**Comment by madebr** [comment]
While checking my mails, I received this: https://github.com/conan-io/tribe/pull/34

**Comment by vdsbenoit** [comment]
What a nice timing!

**Review by SSE4 — APPROVED** [review]

**Pull request merged**
